hazelcast-sql-core/src/main/java/com/hazelcast/sql/impl/calcite/validate/operators/common/AbstractCaseOperator.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright (c) 2008-2020, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.sql.impl.calcite.validate.operators.common;
+
+import com.hazelcast.sql.impl.calcite.validate.HazelcastCallBinding;
+import com.hazelcast.sql.impl.calcite.validate.operators.HazelcastCaseOperator.CaseReturnTypeInference;
+import org.apache.calcite.sql.SqlCall;
+import org.apache.calcite.sql.SqlCallBinding;
+import org.apache.calcite.sql.SqlKind;
+import org.apache.calcite.sql.SqlOperator;
+import org.apache.calcite.sql.SqlSyntax;
+import org.apache.calcite.sql.SqlWriter;
+import org.apache.calcite.sql.fun.SqlCaseOperator;
+import org.apache.calcite.sql.type.InferTypes;
+
+import static com.hazelcast.sql.impl.calcite.validate.operators.HazelcastReturnTypeInference.wrap;
+
+public abstract class AbstractCaseOperator extends SqlOperator implements HazelcastOperandTypeCheckerAware {
+
+    protected AbstractCaseOperator() {
+        super(SqlCaseOperator.INSTANCE.getName(), SqlKind.CASE, SqlCaseOperator.INSTANCE.getLeftPrec(), true, wrap(new CaseReturnTypeInference()),
+                InferTypes.RETURN_TYPE, null);
+    }
+
+    @Override
+    public SqlSyntax getSyntax() {
+        return SqlCaseOperator.INSTANCE.getSyntax();
+    }
+
+    @Override
+    public void unparse(SqlWriter writer, SqlCall call, int leftPrec, int rightPrec) {
+        writer.print("CASE");
```

Comment:
What is that? We should never add an arbitrary code anywhere. The proper way is either to copy-paste the source code from Calcite's implementation, or do not override the method at all (depending on our needs).